File: State/cpp/ON.cpp
#include "ON.h"
#include "OFF.h"

ON::ON(){
	cout << "   ON-ctor ";
}

ON::~ON(){
	cout << "   dtor-ON\n";
}

void ON::off(Machine *m){
	cout << "   going from ON to OFF";
	m->setCurrent(new OFF());
	//delete this;
}

File: State/cpp/java/src/dp/state/sm/ON.java
package dp.state.sm;

public class ON extends State {
	
	// Constructor
	public ON(){
		super();
		System.out.println("   ON-constructor ");
	}
	
	public void off(Machine m){
		System.out.println("   going from ON to OFF");
		m.setCurrent(new OFF());
	}

}